### ISSUE TITLE
widgetsnbextension can only be installed with pip

### ISSUE THREAD

**Issue by jdemeyer** [ipython/ipywidgets#718]
The **only** way to have a working installation of `widgetsnbextension` is to use `pip`.

The usual `python setup.py install` does not work, since that invokes `setuptools` (although `setup.py` seems to be written at some point with a conditional import of `setuptools`) and `setuptools` does not handle `data_files` correctly.

Note that it's not a problem for me to use `pip` but it's at least unexpected that `python setup.py install` does not work. So either this should be fixed or detected such that an error can be raised.

CC @embray @SylvainCorlay

**Comment by jdemeyer** [comment]
Even just adding a warning "this package cannot be installed using `python setup.py install`" in the install docs would already help.

**Comment by embray** [comment]
I don't think that's a quite accurate description of the problem, because as I've explained several times `pip` always uses setuptools.

**Comment by jdemeyer** [comment]
Right, the problem is not the `setuptools` module, the problem is the default behaviour of the `setup()` function of `setuptools`.

**Comment by embray** [comment]
Anyways I don't think this is a problem for ipywidgets.

**Comment by Carreau** [comment]
> python setup.py install

This should not be used and has not been really supported for a long time. It will be used internally by pip. If you see anywhere in our docs a direct mention to `python setup.py install` please point the to us so that we can remove it. The problem with a warning of "don't use XXX" is that user will remember XXX, and do XXX – from experience from other IPython options. Especially if the line can be copy/pasted in a terminal.

We'll likely try to add a stricter check in many setup.py to be sure they are ran by pip.

**Comment by jdemeyer** [comment]
> We'll likely try to add a stricter check in many setup.py to be sure they are ran by pip.

If possible, that would be a great idea.

**Comment by malatiyyah** [comment]
I have the same problem

**Comment by jdemeyer** [comment]
I guess the problem is that eggs are not supported. I solved that in cysignals with a fake `bdist_egg` command: https://github.com/sagemath/cysignals/blob/master/setup.py#L185

**Comment by malatiyyah** [comment]
do you mean I change the content of the file?

**Comment by embray** [comment]
Just don't use `./setup.py install` directly.  The problem is that it does the wrong thing for legacy support reasons.

I think in the past i had a good proposal for how to fix this in setuptools, but didn't follow through on it.  The best thing to do is to install with pip though.
